project/webapp/templates/thread.html

```diff
@@ -90,4 +94,33 @@
     //
     // if (cw.ws.readyState == WebSocket.OPEN) cw.ws.onopen();
 </script>
+
```

Comment:
All of this logic should be moved to the thread.js file. Use backbone.js events to attach and handle events. So instead of being in separate script tags, the code should be using the backbone view event structure (http://backbonejs.org/#View-events) .